src/Entities/Player.cpp

```diff
@@ -146,6 +146,22 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 
 
 
+void cPlayer::InitializePlayer()
```

Comment:
Why not utilise the magic of vtables and overload Init for players?


On 13 Feb 2016, at 18:42, "LogicParrot" <notifications@github.com> wrote:

> In src/Entities/Player.cpp:
> 
> > @@ -146,6 +146,22 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
> >  
> >  
> >  
> > +void cPlayer::InitializePlayer()
> The distinction is because we need to call call GetWorld()->AddPlayer(this); instead of calling GetWorld()->AddEntitythis);.
> The the old code did something hacky instead:
> 1. For every player, it calls initialize which calls world->addEntity
> 2. Then, it calls world->AddPlayer, but to avoid the player being added twice, there's a addEntityIfNotPresent inside world->AddPlayer.
> 
> So basically the old code adds the player twice, but aborts the second add.
> 
> —
> Reply to this email directly or view it on GitHub.
>